### PR TITLE
Profile Receive Text Messages for Notifications

### PR DESCRIPTION
**Pull request by elwoodva**
## Description

The VET360 isTextPermitted checkbox has been implemented for the mobile phone on the profile field.  The mockups are located here.

https://adhoc.invisionapp.com/share/FSTDOO0WUHC#/screens/378179664

Note: The mockup shows a checkbox icon on the success message but that is not in adherence with the current alert message design pattern.  The same is true for the size of the font on the label next to the checkbox which is why the actual text wraps.  The mockup does not show the success message going away after 3 seconds but that was discussed and approved, otherwise it would just remain on the screen even if the Veteran updated something else on the Profile page.  One last thing is that there is no mockup of the checkbox being on the edit modal for the mobile phone number but that was added.

## Testing done

NOTE: Testing this may require a local build because the user must be enrolled in VA health care.  That can be simulated locally by editing src/applications/hca/actions.js and changing line 11 from false to true.  Be sure not to check that change in.

- [ X ] yarn test:unit
- [ X ] yarn lint
- [ X ] yarn test:e2e
- [ X ] yarn test:accessibility
- [ X ] Test with local build in browser as a logged in VA Healthcare Veteran to see that it behaves as expected.

## Screenshots

The profile page will display the checkbox underneath the Mobile Phone number with the current VET360 value of the isTextPermitted flag.

Screenshot 2019-09-17 at 9 00 10 AM

---
After the checkbox is checked, the mobile phone number and the checkbox are temporarily replaced by the information message letting the Veteran know their mobile phone number is being saved.

Screenshot 2019-09-17 at 9 10 11 AM

---

After the transaction is saved, a success message will display for 3 seconds.

Screenshot 2019-09-17 at 9 00 30 AM

--- 

After three seconds the success message will go away.

Screenshot 2019-09-17 at 9 00 20 AM

--- 

The receive text message checkbox is also functional on the edit modal for any phone number field that is text enabled which is usually just the mobile phone number.

Screenshot 2019-09-17 at 9 54 02 AM

## Acceptance criteria
- [ ] Checkbox shows up on the profile page underneath the mobile phone number when the user is logged in and is currently enrolled in VA Health care.  The checkbox will also only show up when there is a mobile phone number that has the isTextable flag set.
- [ ] When checkbox is checked, the mobile phone number and the checkbox are replaced temporarily by the information message letting the Veteran know their data is being saved.
- [ ] When the transaction is successfully saved the mobile phone number and checkbox with the new value re-appears and a success message is displayed.
- [ ] After 3 seconds the success message goes away.
- [ ] Checkbox also shows up on the edit modal for the mobile phone number and will be updated when saved just like the rest of the mobile phone number.  No success message is displayed afterwards when it is updated this way.

## Definition of done
- [ X ] Events are logged appropriately
- [ X ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ X ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by elwoodva** [comment]
I added work for Issue #1281 to put the checkbox on the phone number edit modal.

**Comment by ncksllvn** [comment]
I know this is still a WIP but offering some advice about the success message. It's been a long time since I worked on these components, but I think we made the decision to not show success alerts at some point during the profile work, so when a transaction is finished processing and returns from the API with a status indicating success, it's automatically removed from state - https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/vet360/actions/transactions.js#L92. Could be wrong, but you may need to tweak that.

Besides that, I'm sure you already discovered the `isSuccessfulTransaction` util function that accepts a transaction object in order to determine whether a transaction status indicates success. I think these should be all of the pieces you need.

This is looking great so far by the way!

**Comment by elwoodva** [comment]
@ncksllvn If there is not to be a success message then that makes it easy and I hope we can go with that.  It's on my requirements mockup so I was figuring out how to do it.

I did see the isSuccessfulTransaction call but by the time I was ready to check that, I no longer had the transaction...it was undefined.  So I was trying to figure out a way to store it, or get back to it, or something like that.  

I'll ask if we even need it.  I think it would be fine without the success message.  But we'll see what they think.

Thanks for taking a look!  I appreciate your time and advice.

**Comment by ncksllvn** [comment]
We did show success messages initially but later IIRC we considered the messages to be kinda disruptive (at the time, the success alert would show at the top of the Profile, and so we'd have to scroll them to the top to put it into focus) and that just rendering with the updated data was enough user feedback.

Thank you @elwoodva! 😄

**Comment by elwoodva** [comment]
We are supposed to show the success message.  So that has yet to be worked out properly.

**Comment by elwoodva** [comment]
The success message is now triggered at the appropriate time when the mobilePhone transaction that is originated when the Veteran clicks on the checkbox and it stays on the page for 3 seconds.

**Comment by rianfowler** [comment]
Please tag the front end review group in a comment when this has an approval from a team member. We recommend using the [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) until PRs have an approval from a team member. You can see the review guidelines for this repo [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md)

Thanks!

**Review by talbotmj — APPROVED** [review]
I've reviewed the code (with Elwood and on my own) and I have also built and ran the code locally. The functionality is behaving as specified and the screens match the mock-ups received. I am approving as a team member - first review.

**Comment by accbjt** [comment]
@elwoodva can you add some screenshots and acceptance criteria to the description above. I don't think you updated that yet. When reading the PR other developers don't know what to test and what to look for. Please add a step by step instruction on how you might test the new feature.

**Comment by elwoodva** [comment]
My bad.  Yes, I will do that.

**Review by rianfowler — APPROVED** [review]

**Review by accbjt — APPROVED** [review]

**Review by ncksllvn — COMMENTED** [review]
👋 Hey folks, if you could see my [comment](https://github.com/department-of-veterans-affairs/vets-website/pull/10730) on another open pull request for some really exciting work, I recommend coming to a stopping point on this soon, identifying remaining work items, and mapping those items into tickets. For example -

- [x] Implement foundation for text messaging on mobile devices (could be associated with the work done in this PR)
- [ ] Implement user state X
- [ ] Implement user state Y
- [ ] Error-handling
- [ ] Add E2E test
- [ ] Implement Google Analytics
- [ ] Probably a lot more!

This will help with sharing the remaining work between engineers on the same team and keep pull requests focused to a certain piece of the application so that code review is easier and more valuable. Just a suggestion!

File: src/platform/user/profile/vet360/containers/ReceiveTextMessages.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/containers/ReceiveTextMessages.jsx]: Alright!  Will do.  Thanks for spotting that.

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/components/MobilePhone.jsx]: The feature flags were just recently brought to our attention.  I think you are probably right and that would be the right way to go.  I will discuss with our team.

File: src/platform/user/profile/vet360/containers/ReceiveTextMessages.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/containers/ReceiveTextMessages.jsx]: Wait, actually it already is applied in MobilePhone.jsx which goes to your point about switching to a feature flag I think.

**Pull request merged**
